packages/components/src/components/combobox/combobox.component.ts

```diff
@@ -0,0 +1,963 @@
+  private setSelectedOption(option: SynOption | undefined) {
+    const allOptions = this.getAllFilteredOptions();
+
+    // Clear existing selection
+    allOptions.forEach(el => {
+      // eslint-disable-next-line no-param-reassign
+      el.selected = false;
+    });
+
+    // Set the new selection
+    if (option) {
+      // eslint-disable-next-line no-param-reassign
+      option.selected = true;
+      this.lastOptionValue = option.value;
+    }
+
+    // Update selection, value, and display label
+    this.selectionChanged();
+  }
+
+  // This method must be called whenever the selection changes.
+  // It will update the selected options cache, the current value, and the display value
+  private selectionChanged() {
+    // Update selected options cache
+    this.selectedOption = this.getAllFilteredOptions().find(el => el.selected);
+
+    // Update the value
+    this.value = this.selectedOption?.value ?? this.displayInput.value;
+
+    // Update validity and display label
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.updateComplete.then(() => {
+      this.displayLabel = this.selectedOption?.getTextLabel() ?? this.displayInput.value;
+      this.formControlController.updateValidity();
+    });
+  }
+
+  private handleInvalid(event: Event) {
+    this.formControlController.setValidity(false);
+    this.formControlController.emitInvalidEvent(event);
+  }
+
+  @watch('filter', { waitUntilFirstUpdate: true })
+  handleFilterChange() {
+    this.createComboboxOptionsFromQuery(this.value);
+  }
+
+  @watch('disabled', { waitUntilFirstUpdate: true })
+  handleDisabledChange() {
+    // Disabled form controls are always valid
+    this.formControlController.setValidity(this.disabled);
+
+    // Close the listbox when the control is disabled
+    if (this.disabled) {
+      this.open = false;
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      this.handleOpenChange();
+    }
+  }
+
+  @watch('value', { waitUntilFirstUpdate: true })
+  handleValueChange() {
+    // set the display label here in case of the value was set via property only
+    this.displayLabel = this.value;
+    this.createComboboxOptionsFromQuery(this.value);
+    this.setCurrentOption(null);
+  }
+
+  @watch('open', { waitUntilFirstUpdate: true })
+  async handleOpenChange() {
+    if (this.open && !this.disabled) {
+      if (this.filteredOptions.length === 0) {
+        // Don't open the listbox if there are no options
+        this.open = false;
+        this.emit('syn-error');
+        return;
+      }
+
+      // Show
+      this.emit('syn-show');
+      this.addOpenListeners();
+
+      await stopAnimations(this);
+      this.listbox.hidden = false;
+      this.popup.active = true;
+
+      const { keyframes, options } = getAnimation(this, 'combobox.show', { dir: this.localize.dir() });
+      await animateTo(this.popup.popup, keyframes, options);
+
+      this.emit('syn-after-show');
+    } else {
```

Comment:
You could omit the `else` here by returning in the `if` statement, which could make the code easier to read.